runtime/common/memorykind.go

```diff
@@ -110,6 +110,7 @@ const (
 
 	// Cadence Types
 	MemoryKindCadenceOptionalType
+	MemoryKindCadenceRestrictedType // Deprecated: removed in v1.0.0
```

Comment:
Include "Deprecated" in the name, to match with the Go type name (`DeprecatedRestrictedType`)